mullvad_module.sh

```diff
@@ -50,23 +50,23 @@ VPN_CODES=("${VPN_LOCATIONS[@]}")
 VPN_CODES+=("${COUNTRY_CODES[@]}")
 VPN_LOCATIONS+=("${COUNTRIES[@]}")
 
+MULLVAD_STATUS=$(mullvad status | cut -d' ' -f3)
 
 mullvad_report(){
 # continually reports connection status
 
-	mullvad_status=$(mullvad status)
-	ip_address=$(echo $mullvad_status | grep -oe [0-9].*: | cut -f1 -d:)
+	ip_address=$(mullvad status | cut -d' ' -f7 | cut -d':' -f1)
 
-	if echo $mullvad_status | grep -q 'Connected'; then
+	if [ "$MULLVAD_STATUS" == "Connected"  ]; then
 		if hash geoiplookup 2>/dev/null; then
-			country=$(geoiplookup $ip_address | grep "GeoIP Country" | cut -d' ' -f4 | cut -d, -f1)
-			city=$(geoiplookup $ip_address | grep "GeoIP City" | cut -d, -f5 | sed 's/ //')
+			country=$(geoiplookup "$ip_address" | head -n1 | cut -c24-25)
+			city=$(geoiplookup "$ip_address" | cut -d',' -f5 | sed -n '2{p;q}' | sed 's/ //')
```

Comment:
removing more `grep` for speed, line 63 could also be `city=$(geoiplookup "$ip_address" | head -n2 | tail -n1 | cut -d',' -f5 | sed 's/ //')`, it's about the same speed